ACD-992 BigCommerce (#79)
Actindo-AG/Documentation: BigCommerce/Integration/01_ManageBigCommerceConnection.md

> # Manage BigCommerce connection

Reviewer: Denke bitte daran, noch die Related content Links in Integration, Troubleshooting and User interface Datei einzufügen.

Author: done

> *Actindo* features a special installation app in the *BigCommerce App Store*. The *Actindo Core1 DOP connector* app setups and authorizes a connection automatically. 

Reviewer: ```suggestion
*Actindo* features a special installation app in the *BigCommerce App Store*. The *Actindo Core1 DOP connector* app sets up and authorizes a connection automatically. 
```
set up als Verb schreibt man getrennt, m.W.: https://www.merriam-webster.com/dictionary/setup

Author: done

> Note, if possible, you should not delete the app in *BigCommerce* after creating the connection. Although the deleting will not delete the connection in *Omni-Channel* itself, but it will invalidate the access token you have created during connecting the app. If the app was deleted by mistake, either reconnect the app or enter a valid access token manually to enable the connection again.

Reviewer: Wenn die Kunden das sowieso nicht machen sollten, wollen wir es erwähnen? Ich finde, es klingt nicht "wichtig" genug ("Note, if possible,..."). Vielleicht ein Info-Box (oder sogar Caution) mit der Info, was sie tun sollten, FALLS es aus Versehen gelöscht wird? 
Umformulierungsvorschlag: Although deleting the connection the app will not delete the connection..., it will invalidate the access token... (Sonst gibt es zwei Konzessive-Sätze, finde ich: Although ..., aber...)

Note that you can use the *Actindo Core1 DOP connector* app for one connection only. If you have more than one store in *BigCommerce*, you can <!--- can oder must?-->create the first connection automatically with the *Actindo Core1 DOP connector* app. For all other stores, you must create the connection manually.

> 

Reviewer: Frage geklärt? "... you can <!--- can oder must?-->create the first connection automatically..."
Evtl. Kommentar für Fachreviewer hinterlassen, damit man die Frage besser sehen kann.

## Create BigCommerce connection automatically 

Create the connection using the *BigCommerce* driver. You will not be able to configure any other settings until the connection is established.

### Prerequisites

- You have a *BigCommerce* customer account to log in to *BigCommerce*.
- You have created a *BigCommerce* store. 

> - In the *Actindo Core1 Platform*, you have permission to create connections in *Omni-Channel*.

Reviewer: Braucht man wirklich eine spezielle Rolle/Genehmigung, um eine Verbindung erstellen zu können? Ich finde keine Option dazu in the ActindoUserGroup, s. Screenshot unten (aber ich bin in meinen Sandbox Administratorin). Bisher haben wir es bei den Verbindungen nicht erwähnt, daher bin ich unsicher, ob es so ist. Evtl. kannst du auch Oli dazu fragen.

Wenn ja, vielleicht ein Verweis hier auf die Core1-Doku (ManageUserGroup) hinzufügen?

![image](https://github.com/Actindo-AG/Documentation/assets/99268392/2f819b27-e804-4550-ba05-6b497f538e9a)

### Procedure

> 1. Log in to your *Actindo Core1* instance, on which you run the *Omni-Channel* module. To verify the successful creation later, it is important to be logged in.

Reviewer: ```suggestion
1. Log in to your *Actindo Core1* instance on which you run the *Omni-Channel* module. To verify the successful creation later, it is important to be logged in.
```
Ich bin der Meinung, da braucht man kein Komma, da es sich um eine notwendige Info handelt (s. defining vs. non-defining relative clauses: https://dictionary.cambridge.org/grammar/british-grammar/relative-clauses-defining-and-non-defining)

Author: ok

2. In a new browser window, open the *BigCommerce App Store* (https://www.bigcommerce.com/apps/) and log in to your *BigCommerce* account using the [Log in] button in the upper right corner.   
  The *BigCommerce Actindo-sandbox* is opened.

   ![BigCommerce](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/BigCommerceEntryScreen.png "[BigCommerce]")

> 2. Click the *Apps* folder.

Reviewer: ```suggestion
3. Click the *Apps* folder.
```
Ab hier, Nummerierung anpassen.

Author: ok

3. Click the *Marketplace* folder and select the *Actindo Core1 DOP Connector* app.   
  The *Actindo Core1 DOP Connector* app entry is displayed.

   ![Select app](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/SelectApp.png "[Select app]")

 4. Click the *Actindo* logo.   
  The app is selected. Further information is provided. 

    ![Actindo Onboarding app](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/DownloadOnboardingApp.png "[Actindo Onboarding app]")

> 

Reviewer: Nur so am Rand: Die Actindo App ist unter "My Draft Apps" und da kommt die Info (Disclaimer), das es nicht "approved" ist. Das sieht komisch aus. Geht es nicht anders? Also, ist es einfach noch nicht approved?
Sollte man hier was dazu erwähnen, um potenzielle Bedenken beim Kunden zu beheben? ;)

Author: done

5. Click the [Install] button.  
  The site for confirming the access is opened.

   ![Confirm access](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/ConfirmAccess.png "[Confirm access]")

  
6. Click the [Confirm] button.   
  The *BigCommerce installation* window is opened.

   ![BigCommerce installation](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/EnterWorkspaceURL.png "[BigCommerce installation]")

> 7. Enter the workspace URL of the instance, on which your *Omni-Channel* module is running and click the [START SETUP] button.

Reviewer: ```suggestion
7. Enter the workspace URL of the instance on which your *Omni-Channel* module is running and click the [START SETUP] button.
```
Defining clause. (Vgl. zum letzten Satz, mit Komma, da es "non-defining/explanatory clause", also extra Info, ist.)

Author: done

   If the installation was successful, the *BigCommerce connected* window is displayed.
   In your instance, a new *Omni-Channel* connection is created, which is connected to your *BigCommerce* shop.   
   
   ![Connected to BigCommerce](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/ConnectedToBigCommerce.png "[Connected to BigCommerce]")

8. Change to your *Actindo Core1* instance and select *Omni-Channel > Settings > Tab CONNECTIONS*.   
    - The new connection has been created, synchronized, and activated.   

>     - The credentials of the connection have been automatically created.

Reviewer: credentials are automatically created oder eher filled in...?

Author: nö

    - All related standard attributes, attribute sets, and attribute groups have been created. If required, you can adapt these objects to your needs later using the *DataHub* or *PIM* module.

>     - All relevant APIs settings in *BigCommerce* are automatically created. See the *Store-level API account* in the *BigCommerce* app under *Settings > API*.

Reviewer: ```suggestion
    - All relevant API settings in *BigCommerce* are automatically created. See the *Store-level API account* in the *BigCommerce* app under *Settings > API*.
```

Author: done

   ![New Omni-Channel connection](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/OmniChannelConnectionCreated.png "[New Omni-Channel connection]")

   ![Connection credentials](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/ConnectionCredentials.png "[Connection credentials]")

> 

Reviewer: Beim zweiten Screenshot fehlt teilweise der graue Rahmen (vermutlich beim Cropping).
Wir haben i.d.R. keine zwei Screenshots untereinander. Evtl. zweite Screenthot woanders oder eine kurze Erklärung dazwischen?

Author: done

### Prerequisite

- You have already created a connection automatically.<!---Janik stimmt das, ist das wirklich eine Voraussetzung?-->
- You have both the *Store hash* and the *Access token* provided by *BigCommerce* at your disposal.
- In the *Actindo Core1 Platform*, you have permission to create connections in *Omni-Channel*.

### Procedure

*Omni-Channel > Settings > CONNECTIONS*

![Omni-Channel connections](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/Connections.png "[Omni-Channel connections]")

1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button in the bottom right corner.   
    The *Create connection* view is displayed.

    ![Add connections](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/AddConnection.png "[Add connections]")

2. Enter a name for the connection in the *Name* field.

3. Click the *Driver* drop-down list and select the *BigCommerce* driver.  
    The *Credentials* section is displayed below the drop-down list. 

4. Enter the hash key of your store in the *Store hash* field and the access token in the
  *Access token* field. Both data is provided by *BigCommerce*.

> 

Reviewer: Vielleicht hier ein Screenshot mit den Credentials-Feldern?

Author: done

5. Click the [SAVE] button.

> 

Reviewer: Kommt hier keine Bestätigungsmeldung/Pop-Up-Fenster? Sync-Meldung?

Author: done

6. Click the ![Back](../../Assets/Icons/Back02.png "[Back]") [Back] button and check whether the new connection has been synchronized and activated.   

>     - The new connection has been created, synchronized, and activated.   

Reviewer: In den anderen Connections, die Liste von Verbindungen ist direkt angezeigt (mit oder ohne Bestätigungsmeldung), daher finde ich es unüblich, dass man auf Back drücken (kann auch natürlich sein). Ich finde aber auch merkwürdig, dass man sagt "check whether the connection is activated" und dass man direkt darunter, man behauptet, dass es tatsächlich passiert ist. Vielleicht der erste Satz (Click the Back button and check whether...) umformulieren? Oder 5. und 6. zusammenfügen? (So was? s. unten)

5. Click the [SAVE] button.
   The connection has been created and synchronized... (+ zusäztliche Info) 
      
   Screenshot
   
   Info If necessary, click the Back button...

Author: done

    - All related standard attributes, attribute sets, and attribute groups have been created. If required, you can adapt these objects to your needs using the *DataHub* or *PIM* module later.

>     - All relevant APIs settings in *BigCommerce* are automatically created. See the *Store-level API account* in the *BigCommerce* app under Settings > API.

Reviewer: ```suggestion
    - All relevant API settings in *BigCommerce* are automatically created. See the *Store-level API account* in the *BigCommerce* app under Settings > API.
```

Author: done

## Configure BigCommerce connection 

Configure the *BigCommerce* connection after you have created it.   

In the *Omni-Channel* settings, you can configure the *Variant option type* that defines the way in which the offer variants are presented in the shop.   
By the *Variant option type* setting, you can define how your customers can select a product variant in *BigCommerce*. Since *BigCommerce* allows multiple types for variant options, the driver needs a specification how to determine the preferred type for a variant option. You can define variant option types such as dropdown, radio button, or rectangle as default for your offers in the *Variant option type* setting.   
For specific offer types, you can use the color-swatch type that is determined in a specific way. For detailed information, see [Configure color-swatch variant option](./04_ManageProductData.md#configure-color‐swatch-variant-option).

With the *Order filter* setting, you can filter the orders that are to be imported from *BigCommerce* to *Omni-Channel*.  

> Additionally, if you have worked with *BigCommerce* before starting with the *Actindo Core1 Platform*, you can define the date or ID, from which existing orders should be imported.

Reviewer: ```suggestion
Additionally, if you have worked with *BigCommerce* before starting with the *Actindo Core1 Platform*, you can define the date or ID from which existing orders should be imported.
```

Author: nö

> The *Offers* setting is only relevant for the first initial import of products from *BigCommerce*. You can specify that only those attributes are to be imported, that are mapped to the offer by the ETL data mapping. This allows you to maintain only certain attributes.

Reviewer: ```suggestion
The *Offers* setting is only relevant for the first initial import of products from *BigCommerce*. You can specify that only those attributes that are mapped to the offer by the ETL data mapping are to be imported. This allows you to maintain only certain attributes.
```

Author: nö

#### Prerequisites

- In the *Actindo Core1 Platform*, you are permitted to create connections in *Omni-Channel*.

> - A *BigCommerce* connection has been created.

Reviewer: Die zweite Voraussetzung enthält die erste, also, dass man Verbindungen erstellen darf, daher würde ich die erste Voraussetzung weglassen. (Wir haben es so irgendwann entschieden, weil es logischer und "sparsamer" ist, oder?)

#### Procedure

*Ommi-Channel > Settings > Tab CONNECTIONS*

![Omni-Channel connections](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/OmniChannelConnectionCreated.png "[Omni-Channel connections]")

 1. Click the *BigCommerce* connection you want to edit in the list of connections.   

>    The *Credentials* and the *Settings* tab are displayed in the *Edit connection* view. By default, the *Credentials* tab is selected.

Reviewer: ```suggestion
   The *Edit connection* view is displayed. By default, the *Credentials* tab is selected.
```
So fänd ich es genauer beschrieben. Der Settings tab is sichtbar, weil er da ist, aber er ist (dessen Inhalt) nicht wirklich angezeigt.

Author: wie schön, dass ich das alles irgendwo weg kopiert habe

    ![Connection credentials](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/ConnectionCredentials.png "[Connection credentials]")

> 

Reviewer: Zu den Credentials: Sollte man diese Credentials nicht lieber anonymisieren?

2. Click the *Settings* tab.

> 

Reviewer: The *Settings* tab is displayed. By default, the *Variant option type* setting is selected.

Screenshot 

Result fehlt. Also, ich finde, man kann 2 und 3 teilweise zusammenfügen.

> 3. Click the *Variant option type* entry.   

Reviewer: Click the drop-down menu and select... (also, 3 und 4)

>    The *Variant option type used for variants* dropdown is displayed.

Reviewer: Wir nennen es drop-down list. Wenn es in BigCommerce "dropdown" heißt (in BigCommerce UI), würde ich als dropdown da lassen (aber nur, wenn wir uns auf das BigCommerce UI beziehen). Für Actindo-bezogene UI würde ich bei unserer Schreibweise bleiben...

    ![Variant option type](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/VariantOptionType.png "[Variant option type]")

4. Select the *Variant option type* you want to use.
   The following option types are available:

>    | Dropdown | Meaning   

Reviewer: Zu den Layout: Ich finde die Tabelle an sich nicht schlecht, aber bei den "available options" nach einer Drop-down-Liste haben wir immer Aufzählung gemacht (also, bullet points in Markdown)...

Author: d

   |-----|-------   

>    |Dropdown| An offer variant is selected in a dropdown

Reviewer: ```suggestion
   |Dropdown| An offer variant is selected in a drop-down list / menu (?).
```

Author: Done

>    |Rectangles | An offer variant is selected in a small rectangle, that contains the offer variant value. For example:![Rectangle](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/VariantOptionRectangle2.png "[Rectangle]")

Reviewer: ```suggestion
   |Rectangles | An offer variant is selected in a small rectangle that contains the offer variant value. For example:![Rectangle](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/VariantOptionRectangle2.png "[Rectangle]")
```

Author: Done

    |Radio buttons | An offer variant is selected with a radio button.   

> 5. Click the *Order filters* option.   

Reviewer: ```suggestion
5. Click the *Order filters* entry.   
```
Vgl. Schritt 3.
Wir haben i.d.R. "menu entry" genannt. Dann als Result "The order filters setting is displayed on the right side.

Author: Geht information verloren

   A list of order status is displayed.

   ![Order filters](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/OrderFilters.png "[Order filters]")  

6. Enable the toggles at the order statuses you want to import to the *Actindo Core1 Platform*. For detailed information on the *BigCommerce* order statuses, see the *BigCommerce* documentation under [https://developer.bigcommerce.com/docs/rest-management/orders/order-status#get-all-order-statuses](https://developer.bigcommerce.com/docs/rest-management/orders/order-status#get-all-order-statuses).

> 7. If desired, enter the first order ID you want to import in the *Import only orders with an ID as of* field. This option is only relevant if you have orders in *BigCommerce* that should not be imported to *Actindo* in general.

Reviewer: Im UI steht "Import only orders with an ID greater than"... Stimmt das? Absichtlich gemacht? (as of ist aber schöner, gebe ich zu)

8. If desired, enter the date as of which you want to import the orders in the *Import only orders created as of* field. The date is valid from 00:00. This option is only relevant if you have orders in *BigCommerce* that should not be imported to *Actindo* in general.

> 9. If desired, request the import of orders that have been already archived by enabling the *Import archived order* toggle.

Reviewer: Evtl. andersum, wie die anderen Anweisungen?
If desired, enable the *Import archived order* toggle to request the import...

Author: Done

> 10. Click *Offers* option.

Reviewer: option -> menu entry

> 

Reviewer: Offers menu entry kommt vorher in den Screenshots nicht vor. Muss man was voreinstellen oder installieren (plug-in)? Wieso wird es nicht angezeigt?

Result fehlt: The offers setting is displayed on the right side.

Author: Done

   ![Offers option](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/OfferOption.png "[Offers option]")

11. Enable the *Only push mapped attributes* toggle, if you want to import only those attributes that are mapped to the offer by the ETL data mapping. 

> 12. Click the [Save] button.   

Author: Ist es aber nicht

  The connection settings are applied.

> ## Manage ETL mappings 

Reviewer: Ich finde, das passt inhaltlich hier nicht ganz. Ich denke, man könnte eine 05_ManageETLMappings.md erstellen und ein wenig erweitern?

Author: esgibt nicht mehr zu berichten

> Create ETL mappings to map the *BigCommerce* attributes for subsequent processes in your *Actindo Core1 platform*. During connecting *BigCommerce* with *Actindo*, the *BigCommerce* attribute sets have been automatically created in the *DataHub* module.   

Reviewer: ```suggestion
Create ETL mappings to map the *BigCommerce* attributes for subsequent processes in your *Actindo Core1 platform*. While connecting *BigCommerce* with *Actindo*, the *BigCommerce* attribute sets have been automatically created in the *DataHub* module.   
```
Oder During the connection of BigCommerce with... 
https://dictionary.cambridge.org/de/grammatik/britisch-grammatik/during

Author: hat mir meine Übersetzung so beigebracht

[!! Connections (BigCommerce)](../UserInterface/01a_Connections.md)

# Manage BigCommerce connection

*Actindo* features a special installation app in the *BigCommerce App Store*. The *Actindo Core1 DOP connector* app sets up and authorizes a connection automatically. 
Note, if possible, you should not delete the app in *BigCommerce* after creating the connection. Although deleting will not delete the connection in *Omni-Channel* itself, it will invalidate the access token you created when connecting to the app. If the app was deleted by mistake, either reconnect the app or enter a valid access token manually to enable the connection again.

> Note that you can use the *Actindo Core1 DOP connector* app for one connection only. If you have more than one store in *BigCommerce*, you can <!--- Jannik, can oder must?-->create the first connection automatically with the *Actindo Core1 DOP connector* app. For all other stores, you must create the connection manually.

Reviewer 2: Das ist nicht ganz korrekt, die App kann nur eine Connection für einen Store in BigCommerce erzeugen. Hat der Kunde mehrere (unterschiedliche) Stores, kann er die App nutzen um mehrere Connections zu erzeugen. Wichtig dabei ist, dass der Storehash unterschiedlich ist. 
Wurde bereits eine Connection mit dem Storehash erzeugt, egal ob durch die App oder manuell, erzeugt die App keine neue Connection, sondern aktualisiert die mit dem passenden Storehash.

Author: Done

    - The new connection has been created, synchronized, and activated.   

       ![New Omni-Channel connection](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/OmniChannelConnectionCreated.png "[New Omni-Channel connection]")

    - The credentials of the connection have been automatically created.   

      ![Connection credentials](../../Assets/Screenshots/Channels/Settings/Connections/BigCommerce/ConnectionCredentials.png "[Connection credentials]")

    - All related standard attributes, attribute sets, and attribute groups have been created. If required, you can adapt these objects to your needs later using the *DataHub* or *PIM* module.
    - All relevant API settings in *BigCommerce* are automatically created. See the *Store-level API account* in the *BigCommerce* app under *Settings > API*.

## Create BigCommerce connection manually

After automatically creating a connection for one *BigCommerce* store, you may need to manually create additional connections for additional stores, if applicable.      
If you create the connection manually, you have no functional limitations compared to the automatic creation.

### Prerequisite

> - You have already created a connection automatically.<!---Janik stimmt das, ist das wirklich eine Voraussetzung?-->

Reviewer 2: Nein das ist keine Voraussetzung

Author: Done
